examples/graph_sage_unsup.py

```diff
@@ -4,40 +4,24 @@
 import torch.nn as nn
 import torch.nn.functional as F
 from sklearn.linear_model import LogisticRegression
-from torch_cluster import random_walk
 
 import torch_geometric.transforms as T
 from torch_geometric.datasets import Planetoid
-from torch_geometric.loader import NeighborSampler as RawNeighborSampler
+from torch_geometric.loader import LinkNeighborLoader
 from torch_geometric.nn import SAGEConv
 
+#from torch_cluster import random_walk
+
 EPS = 1e-15
 
 dataset = 'Cora'
 path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', dataset)
 dataset = Planetoid(path, dataset, transform=T.NormalizeFeatures())
 data = dataset[0]
 
-
-class NeighborSampler(RawNeighborSampler):
-    def sample(self, batch):
-        batch = torch.tensor(batch)
-        row, col, _ = self.adj_t.coo()
-
-        # For each node in `batch`, we sample a direct neighbor (as positive
-        # example) and a random node (as negative example):
-        pos_batch = random_walk(row, col, batch, walk_length=1,
-                                coalesced=False)[:, 1]
-
-        neg_batch = torch.randint(0, self.adj_t.size(1), (batch.numel(), ),
-                                  dtype=torch.long)
-
-        batch = torch.cat([batch, pos_batch, neg_batch], dim=0)
-        return super().sample(batch)
-
-
-train_loader = NeighborSampler(data.edge_index, sizes=[10, 10], batch_size=256,
-                               shuffle=True, num_nodes=data.num_nodes)
+train_loader = LinkNeighborLoader(data, batch_size=256, shuffle=True,
+                                  neg_sampling_ratio=1.0,
+                                  num_neighbors=[10, 10])
```

Comment:
Can we also increase `num_workers` to `6` (or similar)?

Reply:
I tried to use `num_workers = 6` but was getting an error ( on a MacBook Pro, no cuda support). So left it unchanged.